README.md

```diff
@@ -8,10 +8,18 @@ https://github.com/strongloop/loopback-next/issues/1476 for more information.
 
 ![Shopping example overview diagram](example-shopping.png)
 
+## Pre-requisites
+
+Node.js >= 8.9.0 and running instances of a MongoDB and Redis server are
+required for the app to start. The Redis server is used for the shopping cart,
+while MongoDB is used for the rest of the models in the app.
+
+Docker is required for running tests, make sure it is running if you want to run
+the tests.
+
 ## Installation
 
-Make sure you have Node.js >= 8.9.0 installed. Then do the following to clone
-and start the project.
+Do the following to clone and start the project.
```

Comment:
When I start the shopping app I have to run `npm test` to start the redis docker instance, then `npm start` will have a working redis connection.
I think that's a step worth mentioning.

Reply:
The fact that `npm test` starts the Redis docker instance is a "lucky" side affect of `npm test`, not an intended step, and there is no guarantee that this behavior is going to be there forever. The user must ensure they have a running Redis instance using whatever means. Let's not include it.

Reply:
@jannyHou , there is 

```
"docker:start": "./bin/start-dbs.sh",
"docker:stop": "./bin/stop-dbs.sh",
```

in package.json

Reply:
In tutorial : https://loopback.io/doc/en/lb4/Authentication-Tutorial.html#try-it-out , there is a try it out section. we mention the step.

Perhaps as @jannyHou  suggests, we should mention the docker:start here too.

@hacksparrow  ^

Reply:
Added this:

> In case you have Docker installed on your system and don't want to manually install
MongoDB and Redis, you can run `npm run docker:start` to download their images and
start the servers.